activemodel/lib/active_model/translation.rb

```diff
@@ -22,6 +22,8 @@ module ActiveModel
   module Translation
     include ActiveModel::Naming
 
+    mattr_accessor :raise_on_missing_translations, default: false
```

Comment:
```suggestion
    singleton_class.attr_accessor :raise_on_missing_translations
```

We shouldn't use Ruby class variables, they are awful.

Reply:
OK.

Just out of curiosity, why is `mattr_accessor` awful? I searched rails codebase and about 40 places use it. Also isn't it from active support?

Reply:
> Just out of curiosity, why is `mattr_accessor` awful?

See: https://github.com/rails/rails/pull/42451. It uses Ruby class variables (`@@`) which are slow and have wonky semantic, as they leak in sub classes.

For your use case a regular instance variable is faster, simpler.

> Also isn't it from active support?

Yes, but I consider it soft deprecated.

Reply:
Thank you! I learned a lot.